core/metric.go

```diff
@@ -64,6 +67,31 @@ func (n Namespace) Strings() []string {
 	return ns
 }
 
+// getSeparator returns an valid separator from nsPriorityList.
+// Otherwise the core separator is returned.
+func (n Namespace) getSeparator() string {
+	mp := n.toMap()
```

Comment:
That's a really good point. For solving that I use a slice of `Separator`.
```Go
package main

import (
	"fmt"

	"github.com/intelsdi-x/snap/core"
)

var (
	sepSortedList = []Separator{
		Separator{Sep: "/"},
		Separator{Sep: "-"},
		Separator{Sep: "_"},
		Separator{Sep: "㊽"},
		Separator{Sep: "Ä"},
		Separator{Sep: "大"},
		Separator{Sep: "小"},
		Separator{Sep: "ᵹ"},
		Separator{Sep: "☍"},
		Separator{Sep: "ヒ"}}

	sepMap map[string]*Separator
)

type Separator struct {
	Sep  string
	Used bool
}

func init() {
	sepMap = make(map[string]*Separator)
	for i, sep := range sepSortedList {
		sepMap[sep.Sep] = &sepSortedList[i]
	}
}

type Namespace core.Namespace

func main() {
	ns := Namespace(core.NewNamespace("p大Älo-p", "ヒwo㊽rld/", "te☍_st小ᵹ"))
	// ns := core.NewNamespace("p大Älo-p", "wo㊽rld/", "te_st")
	// ns := core.NewNamespace("plop", "world", "test")

	fmt.Println(ns.GetSeparator())
}

func (n Namespace) GetSeparator() string {
	for _, elem := range n {
		for _, r := range elem.Value {
			s := fmt.Sprintf("%c", r)
			if sep, ok := sepMap[s]; ok {
				sep.Used = true
			}
		}
	}
	for _, sep := range sepSortedList {
		if !sep.Used {
			return sep.Sep
		}
	}
	return core.Separator
}
```
At the end, it's using a Map of pointer + a slice of Separator (string+bool).